CMakeLists.txt

```diff
@@ -48,6 +28,9 @@ endif()
 
 ## Add serial library
 add_library(${PROJECT_NAME} ${serial_SRCS})
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    POSITION_INDEPENDENT_CODE ON)
```

Comment:
Minor comment: this make it impossible to compile the library without position. An alternative that still defaults to enable -fPIC unless someone explicitly disable it is to add:
~~~cmake
if (NOT DEFINED CMAKE_POSITION_INDEPENDENT_CODE)
  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
endif()
~~~

before the line `add_library(${PROJECT_NAME} ${serial_SRCS})`

Reply:
Yes I agree this shouldn’t be merged as is with this it’s too invasive.  I picked this out of the ros2 branch we’ve been using.

See comment above.

Reply:
If you build the library as a shared library you don't need this.

Reply:
I picked two commits from @leamas PR #231 (to set the so version) and then set BUILD_SHARED_LIBS ON.

Then tested that out of the box it works fine in a colcon workspace...
But if you do `colcon build --cmake-args "-DBUILD_SHARED_LIBS=OFF" ` then you'll get back to ld errors

```
--- stderr: robotiq_driver                             
/usr/bin/ld: /home/alex/ros/h/robotiq/install/serial/lib/libserial.a(unix.cc.o): warning: relocation against `_ZTVN6serial15SerialExceptionE' in read-only section `.text._ZN6serial15SerialExceptionD2Ev[_ZN6serial15SerialExceptionD5Ev]'
/usr/bin/ld: /home/alex/ros/h/robotiq/install/serial/lib/libserial.a(serial.cc.o): relocation R_X86_64_PC32 against symbol `_ZTVN6serial6SerialE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
```

If I additionally add the suggestion from @traversaro then it seems to work in both cases:
```colcon build --cmake-args "-DBUILD_SHARED_LIBS=OFF" ``` will build a `./install/serial/lib/libserial.a` which can be used without linking errors

@wjwwood any opinion or preference on this?